modules/vmseries/variables.tf

```diff
@@ -208,4 +196,51 @@ variable "diagnostics_storage_uri" {
   description = "The storage account's blob endpoint to hold diagnostic files."
   default     = null
   type        = string
+}
+
+variable "app_insights_settings" {
+  description = <<-EOF
+  A map of the Application Insights related parameters.
+  
+  If the variable is:
+  - not defined - Application Insights will not be created (default behavior)
+  - defined as empty map `{}` - Application Insights will be created based on the default parameters
+  - defined as a map - Application Insights will be created with the defined properties, for any skipped - a default value will be used.
+
+  Available properties are:
+  - `name`                      - (optional|string) Name of the Applications Insights instance. Can be `null`, in which case a default name is auto-generated.
+  - `workspace_mode`            - (optional|bool)   Application Insights mode. If `true` (default), the "Workspace-based" mode is used. With `false`, the mode is set to legacy "Classic".
+  - `metrics_retention_in_days` - (optional|number) Specifies the retention period in days. Possible values are 0, 30, 60, 90, 120, 180, 270, 365, 550 or 730. Azure defaults is 90.
```

Comment:
This value doesn't have default value, either in resource itself since it is a optional value and in module codebase :)